tests/product/image_builder.py

```diff
@@ -0,0 +1,203 @@
+        for installer in installers:
+            installer_instance = installer(self)
+            self.default_keywords.update(installer_instance.get_keywords())
+
+    def _run_installers(self, installers):
+        cluster = self.cluster
+        for installer in installers:
+            dependencies = installer.get_dependencies()
+
+            for dependency in dependencies:
+                dependency.assert_installed(self)
+
+            installer_instance = installer(self)
+            installer_instance.install()
+
+            self.default_keywords.update(installer_instance.get_keywords())
+            cluster.postinstall(installer)
+
+    def _setup_cluster(self, bare_image_provider, cluster_type):
+        installers = None
+        try:
+            installers = self._cluster_types[cluster_type]
+        except KeyError:
+            print '%s is not a valid cluster type. Valid cluster types are %s' % \
+                  (cluster_type, ', '.join(self._cluster_types.keys()))
+            exit(1)
+
+        config_filename = ConfigurableCluster.check_for_cluster_config()
+
+        if config_filename:
+            self.cluster = ConfigurableCluster.start_bare_cluster(
+                config_filename, self,
+                StandalonePrestoInstaller.assert_installed)
+        else:
+            try:
+                self.cluster, bare_cluster = DockerCluster.start_cluster(
+                    bare_image_provider, cluster_type)
+
+                # If we've found images and started a non-bare cluster, the
+                # containers have already had the installers applied to them.
+                # We do need to get the test environment in sync with the
+                # containers by calling the following two functions.
+                #
+                # Once that's done, the cluster and test environment is in the
+                # same state it would be as if we'd called _run_installers on
+                # a bare cluster, and we can return.
+                #
+                # We do this to save the cost of running the installers on the
+                # docker containers every time we run a test. In practice,
+                # that turns out to be a fairly expensive thing to do.
+                if not bare_cluster:
+                    self._apply_post_install_hooks(installers)
+                    self._update_replacement_keywords(installers)
+                    return
+            except DockerClusterException as e:
+                print e.msg
+                exit(1)
+
+        # If we got a bare cluster back, we need to run the installers on it.
+        # applying the post-install hooks and updating the replacement
+        # keywords is handled internally in _run_installers.
+        self._run_installers(installers)
+
+        if isinstance(self.cluster, DockerCluster):
+            self.cluster.commit_images(bare_image_provider, cluster_type)
+
+    def _get_expected_image_names(self, cluster_type):
```

Comment:
This is probably more cleanly done as having a dict of cluster_type to (master_image_name, slave_image_name) and returning dict[cluster_type].
You should be returning tuples instead of lists because tuples are immutable.
Also, you should probably extract the teradatalabs/pa_test portion of the name to a constant?